api-scripts/construct_mappreview.py

```diff
@@ -0,0 +1,367 @@
+
+    return {
+        'name': vector_resource['name'],
+        'type': 'vector',
+        'url': url,
+        'bounds': bounds,
+        'vector_type': vector_type,
+        'feature_count': feature_count,
+    }
+
+
+def get_vector_layers_metadata(vector_resources):
+    vector_layers = []
+    for vector_layer in vector_resources:
+        if not vector_layer:
+            continue
+        vector_layers.append(get_vector_layer_metadata(vector_layer))
+    return filter(None, vector_layers)
+
+
+def get_mappreview_metadata(resources, source_files, mappreview_sources=[]):
+    """Get metadata needed to display all resources in the map"""
+    raster_resources = [r for r in resources if r['format'] == 'GeoTIFF']
+    vector_resources = [r for r in resources if r['format'] == 'Shapefile']
+    layers = []
+
+    zip_resource = next((r for r in resources if r['format'] == 'ZIP'), None)
+
+    if zip_resource and source_files:
+        # Look at zip sources for spatial resources and add
+        for shp_source in [s for s in source_files if s.endswith('shp')]:
+            if mappreview_sources and shp_source not in mappreview_sources:
+                continue
+            LOGGER.debug(shp_source)
+            path = shp_source.replace('\\', '/')
+            path_dirname = os.path.dirname(path)
+            path_basename = os.path.basename(path)
+
+            name = path.split('/')[-1]
+
+            base = '/'.join(zip_resource['url'].split('/')[0:-1])
+            base = base.replace('https://storage.cloud.google.com/',
+                                'https://storage.googleapis.com/')
+
+            # Identify a .mvt (preferred) or .geojson layer version of the
+            # vector for mapping onto the globe.
+            url = None
+            for extension in ('.mvt', '.geojson'):
+                possible_url = (
+                    f'{base}/{path_dirname}/'
+                    f'{path_basename.replace(".shp", extension)}')
+                # If we're working with an mvt, we cannot request a HEAD on a
+                # directory, so we need to get the metadata.json file instead
+                url_to_check = possible_url
+                if extension == '.mvt':
+                    url_to_check = f"{possible_url}/metadata.json"
+
+                if requests.head(url_to_check).ok:
+                    url = possible_url
+                    break
+
+            if not url:
+                LOGGER.warning(
+                    f"Could not find web format equivalent for {shp_source}")
+                continue
+
+            # Get metadata file
+            metadata_path_end = path_basename.replace('.shp', '.shp.yml')
+            metadata_url = f'{base}/{path_dirname}/{metadata_path_end}'
+
+            vector_resources.append({
+                'name': name,
+                'metadata_url': metadata_url,
+                'url': url,
+            })
+
+        for tif_source in [s for s in source_files if s.endswith('tif')]:
```

Comment:
```suggestion
        for tif_source in [s for s in source_files if s.endswith('.tif')]:
```
Probably not essential, but same comment as above with shp